src/classes/RD2_ScheduleService_TEST.cls

```diff
@@ -2014,6 +2014,45 @@ public with sharing class RD2_ScheduleService_TEST {
         System.assertEquals(null, pause, 'A past Pause Schedule should not be returned');
     }
 
+    /***
+    * @description Verifies that the schedule service confirms a currently active paused RD
+    */
+    @isTest
+    private static void shouldIndicateIfCurrentlyPaused() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+        RD2_ScheduleService.currentDate = DATE_ESTABLISHED;
+
+        npe03__Recurring_Donation__c rd = getRecurringDonationMonthlyBuilder().build();
+
+        List<RecurringDonationSchedule__c> schedules = new RD2_ScheduleService().buildNewSchedules(rd);
+        schedules.add(buildPauseSchedule(rd.Id, Date.today(), Date.today().addMonths(2)));
```

Comment:
When a pause for the next installment is added, it has the next installment Close Date. For example, if a pause for a monthly RD with Day of Month = 1 is paused in Nov, the pause schedules will have Start Date = End Date = Nov 1. So, if the pause is built for the next installment Opp instead using `Date.today()`, I believe the test would fail.

Reply:
Wouldn't buildPauseSchedule() as used in this test always result in a currently active pause regardless of the cadence of upcoming installments?  I would think that the pause would always begin today and extend for two months, covering whatever installments would otherwise occur during that span.

Reply:
The pause schedules starts on the first paused installment close date and ends on the last paused installment date.